src/go/rpk/pkg/cli/cluster/partitions/status.go

```diff
@@ -89,7 +90,24 @@ investigation. A few areas to investigate:
 			status, err := cl.GetPartitionStatus(cmd.Context())
 			out.MaybeDie(err, "unable to request balancer status: %v", err)
 
-			printBalancerStatus(status)
+			var clusterPartitions []rpadmin.ClusterPartition
+			clusterPartitions, err = cl.AllClusterPartitions(cmd.Context(), true, false)
+			if err != nil {
+				fmt.Printf("unable to query all partitions in the cluster: %v", err)
+			}
+
+			const (
+				secBalancerStatus = "Balancer status"
+				secBrokerDist     = "Broker distribution"
+			)
+			sections := out.NewSections(
+				out.ConditionalSectionHeaders(map[string]bool{
+					secBrokerDist:     true,
+					secBalancerStatus: true,
+				})...,
+			)
+			sections.Add(secBalancerStatus, func() { printBalancerStatus(status) })
```

Comment:
Here I also think we can move the printing logic into the `printBalancerStatus` function